### PR TITLE
Update README and versions for 2.40.0 / 23.11

### PR DESCRIPTION
**Pull request by mc-nv**

### CODE REVIEW

File: docs/user_guide/request_cancellation.md [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 # Request Cancellation
 
-Starting from r23.10, Triton supports handling request cancellation received
+Starting from r23.11, Triton supports handling request cancellation received
```
Comment (rmccorm4) [inline docs/user_guide/request_cancellation.md:31]: I think this should stay 23.10 CC @tanmayv25

Comment (mc-nv) [reply docs/user_guide/request_cancellation.md]: Addressed

File: build.py [hunk 2]
```diff
@@ -1384,10 +1384,10 @@ def add_cpu_libs_to_linux_dockerfile(backends, target_machine):
...
 
 RUN mkdir -p /usr/local/cuda/targets/{cuda_arch}-linux/lib
-COPY --from=min_container /usr/local/cuda-12.2/targets/{cuda_arch}-linux/lib/libcudart.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
-COPY --from=min_container /usr/local/cuda-12.2/targets/{cuda_arch}-linux/lib/libcupti.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
-COPY --from=min_container /usr/local/cuda-12.2/targets/{cuda_arch}-linux/lib/libnvToolsExt.so.1 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
-COPY --from=min_container /usr/local/cuda-12.2/targets/{cuda_arch}-linux/lib/libnvJitLink.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
+COPY --from=min_container /usr/local/cuda/lib64/libcudart.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
+COPY --from=min_container /usr/local/cuda/lib64/libcupti.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
+COPY --from=min_container /usr/local/cuda/lib64/libnvToolsExt.so.1 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
+COPY --from=min_container /usr/local/cuda/lib64/libnvJitLink.so.12 /usr/local/cuda/targets/{cuda_arch}-linux/lib/.
```
Comment (rmccorm4) [inline build.py:1390]: Is `lib64` arch agnostic now? I see previously we had `{cuda_arch}` in the source path.

Comment (rmccorm4) [reply build.py:1390]: Everything else LGTM other than this comment.

Comment (mc-nv) [reply build.py:1390]: This value is set by the alternatives. I haven't seen issue over our latest builds.

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**